extractors/personal_identifiers/location_extraction/code_snippet_common.md

```diff
@@ -0,0 +1,37 @@
+```python
+import spacy
+from typing import List, Tuple
+
+nlp = spacy.load("en_core_web_sm")
```

Comment:
in other common code snippets we have the spacy language as parameter, in some we don't. What is the proffered way?

Reply:
Yes, just moving the loading of the spacy model to outside of the function was a bit shortsightes. I would say it would make sense to have the spacy language as a parameter and then load the model from a kind of Singelton to avoid that the model is loaded multiple times like this: 

```suggestion
import spacy
from typing import List, Tuple

loaded_models = {}
def load_spacy(spacy_model):
    if spacy_model not in loaded_models:  
        loaded_models[spacy_model] = spacy.load(spacy_model)
    return loaded_models[spacy_model]


def location_extraction(text: str, extraction_keyword: str, spacy_model: str = "en_core_web_sm") -> List[Tuple[str, int]]:
    """
    @param text: the input text
    @param extraction_keyword: the label that is assigned to extracted words
    @return: positions of extracted names of persons  
    """
    nlp = load_spacy(spacy_model)
    doc = nlp(text)

    name_positions = []
    for ent in doc.ents:
        if ent.label_ == "GPE" or ent.label_ == "LOC":
            name_positions.append((extraction_keyword, ent.start, ent.end))
    return name_positions
```

Reply:
works for me